Python/codegen.c

```diff
@@ -280,6 +280,14 @@ codegen_addop_noarg(instr_sequence *seq, int opcode, location loc)
 static int
 codegen_addop_load_const(compiler *c, location loc, PyObject *o)
 {
+    if (PyLong_CheckExact(o)) {
+        int overflow;
+        long val = PyLong_AsLongAndOverflow(o, &overflow);
+        if (!overflow && val >= 0 && val < 256 && val < _PY_NSMALLPOSINTS) {
```

Comment:
Is `val < _PY_NSMALLPOSINTS` needed here?

Reply:
Oh, yes, it's matches the behavior of `bytecodes.c` (https://github.com/python/cpython/pull/125972#discussion_r1816437449)